lib/ansible/modules/cloud/vmware/vmware_vcenter_settings.py

```diff
@@ -241,8 +241,12 @@ def ensure(self):
         directory_query_limit_size = self.params['user_directory'].get('query_limit_size')
         directory_validation = self.params['user_directory'].get('validation')
         directory_validation_period = self.params['user_directory'].get('validation_period')
-        mail_server = self.params['mail'].get('server')
-        mail_sender = self.params['mail'].get('sender')
+        mail_server = mail_sender = ""
+        try:
+            mail_server = self.params['mail'].get('server')
+            mail_sender = self.params['mail'].get('sender')
+        except Exception as e:
```

Comment:
Nope! Never do an `except Exception`. This is a bad practice since you catch ALL the potential exception, and we may lose something import :-)

You can instead do something like:

```python
mail_server = self.params.get('mail', {}).get('server')
mail_server = self.params.get('mail', {}).get('sender')
``` 

e.g:
```
>>> a = {}
>>> print(a.get('mail', {}).get('a'))
None
```